src/nile/core/compile.py

```diff
@@ -52,16 +60,21 @@ def compile(
 
 
 def _compile_contract(
-    path, directory=None, account_contract=False, disable_hint_validation=False
+    path,
+    directory=None,
+    cairo_path=None,
+    account_contract=False,
+    disable_hint_validation=False,
 ):
     base = os.path.basename(path)
     filename = os.path.splitext(base)[0]
     logging.info(f"🔨 Compiling {path}")
     contracts_directory = directory if directory else CONTRACTS_DIRECTORY
+    cairo_path = cairo_path if cairo_path else contracts_directory
```

Comment:
Not quite sure I understand the intent behind this. Aren't `--cairo_path` and `--directories` two unrelated concepts?

For example, if I'd want to use nile to compile my `starknet/tokens/*.cairo` files and use libraries I have in `lib/cairo`, I would run `nile compile --directory starknet/tokens --cairo_path lib/cairo`. Would this work as intended?

FWIW, `--cairo_path` is "a list of directories, separated by ":" to resolve import paths." So if you want, you could merge them together instead of overwriting.

Reply:
> For example, if I'd want to use nile to compile my `starknet/tokens/*.cairo` files and use libraries I have in `lib/cairo`, I would run `nile compile --directory starknet/tokens --cairo_path lib/cairo`. Would this work as intended?

It will work as intended.

> FWIW, --cairo_path is "a list of directories, separated by ":" to resolve import paths." So if you want, you could merge them together instead of overwriting.

We are not overwriting one with another, we are just setting `contracts_directory` as the default `cairo_path` when the latter is not provided. This allows removing the **contracts** prefix from all the local imports in your project (cairo_path defaults to the project source directory). As you said, those are separated concepts, so it makes sense to keep them separated, as explained next.

> Not quite sure I understand the intent behind this. Aren't --cairo_path and --directory two unrelated concepts?

`directory` (AKA contracts directory) and `cairo path` are two different concepts (that can be unified following a pattern but are not by default).

`directory` is the folder containing the Smart Contracts. This is not for starknet to process but for nile to discover contracts in the project's scope.

`cairo path`, on the other hand, is a set of directories used by the starknet compiler directly (not by nile) to resolve imports in cairo files.

Note that we can avoid passing `cairo path` if we have a project with a contract importing only from external sources (lib modules). Still, we can't avoid setting `directory` if the contracts folder name is distinct from contracts (src, for example). And we can avoid passing `directory` if the contracts folder is named contracts, but we still need to pass cairo path if we are using a structure of imports requiring this.

Reply:
Thanks for the explanation. All good then 👍